alerts/cluster-network-operator/NoRunningOvnMaster.md

```diff
@@ -0,0 +1,75 @@
+# NoRunningOvnMaster
+
+## Meaning
+
+This [alert][NoRunningOvnMaster] is triggered when there are no
+OVN-Kubernetes master control plane pods
+[Running][PodRunning]
+This is a critical-level alert if no OVN-Kubernetes master control plane pods
+are not running for `10m`.
+
+## Impact
+Networking control plane is not functional. Networking configuration updates
+will not be applied to the cluster.
+Without a functional networking control plane, existing workloads may continue
+to be partially functional,
+but new workloads will not be functional.
+Updates required for functioning Kubernetes services will not be performed.
+
+## Diagnosis
+### Cluster network operator
+Cluster Network operator (CNO) manages the CRUD operations for OVN-Kubernetes
+daemonset.
+Verify the CNO is running:
+
+    oc -n openshift-network-operator get pods -l name=network-operator
+
+Verify the [CNO](https://github.com/openshift/cluster-network-operator/) is
+functioning without error by checking the operator Status.
+You can check the Status of the CNO directly using the following command:
+
+    oc describe co network
+
+Pay attention to any error message reported by Status Conditions of type
+Degraded.
+
+Check the CNO logs for when it is reconciling the daemonset for ovnkube-master.
+
+    PODNAME=$(oc -n openshift-network-operator get pods
+                -l name=network-operator -o name)
+    oc -n  openshift-network-operator logs $PODNAME
+
+A successful reconcile for this daemonset looks like this in the CNO logs:
+
+    I0228 14:48:30.941130       1 log.go:184] reconciling (apps/v1, Kind=DaemonSet)
+                                openshift-ovn-kubernetes/ovnkube-master
+    I0228 14:48:30.960944       1 log.go:184] update was successful
+
+Verify the _DESIRED_ number of daemonsets is equal to the number of Kubernetes
+control plane nodes:
+
+    oc get nodes -l node-role.kubernetes.io/master="" -o name | wc -l
+    oc get ds -n openshift-ovn-kubernetes ovnkube-master
+
+### OVN-Kubernetes master pod
+Understand which container is failing in the OVN-Kubernetes master pod by
```

Comment:
Better to have some pre-condition here, like if you are not seeing the running ovn-kube master pod matches the expected daemonset replica count. And you can start the debugging from the problem ovn-kube master pod.